### PR TITLE
Distribution contains /var/data/tmp even if tmp/ dir is cleared by user

### PR DESCRIPTION
**Pull request by uschi2000**

### CODE REVIEW

**Comment by uschi2000** [comment]
DO NOT MERGE. Needs to get rebased onto develop once #116 merges.

**Comment by uschi2000** [comment]
Fixes #114

File: src/main/groovy/com/palantir/gradle/javadist/tasks/DistTarTask.groovy [hunk 1]
```diff
@@ -40,7 +40,18 @@ class DistTarTask extends Tar {
...
             }
 
-            new File(project.projectDir, "var/data/tmp").mkdirs()
+            // Create var/data/tmp
+            File.createTempDir().with { emptyTempDirBase ->
+                emptyTempDirBase.deleteOnExit()
+                emptyTempDirBase.toPath().resolve("data").resolve("tmp").toFile().mkdirs()
+                from(emptyTempDirBase) {
+                    into "${archiveRootDir}/var"
+                    // TODO(rfink) If users exclude 'data', then the temp dir will not be created. That seems wrong?
```
Comment (tpetracca) [inline src/main/groovy/com/palantir/gradle/javadist/tasks/DistTarTask.groovy:49]: Per https://github.com/palantir/gradle-java-distribution/issues/114#issuecomment-246156725 I think we want to just have it be `var/tmp`, and then you don't have to worry about this.

Comment (markelliot) [reply src/main/groovy/com/palantir/gradle/javadist/tasks/DistTarTask.groovy:49]: Unfortunately var/tmp is already used. There was a discussion about this previously.

**Comment by markelliot** [comment]
I had taken a stab at this on Friday, both of these seem valid, but I had a smaller diff: #119

**Pull request closed without merging**